src/output_writers.jl

```diff
@@ -270,126 +270,215 @@ end
+   len_time_dimension :: Int
+             previous :: Float64
+end
 
-    xF_attr = Dict("longname" => "Locations of the cell faces in the x-direction.", "units" => "m")
-    yF_attr = Dict("longname" => "Locations of the cell faces in the y-direction.", "units" => "m")
-    zF_attr = Dict("longname" => "Locations of the cell faces in the z-direction.", "units" => "m")
+"""
+    NetCDFOutputWriter(model, outputs; interval=nothing, frequency=nothing, filename=".",
+                                   clobber=true, globalattrib=Dict(), outputattrib=nothing, slice_kw...)
 
-    u_attr = Dict("longname" => "Velocity in the x-direction", "units" => "m/s")
-    v_attr = Dict("longname" => "Velocity in the y-direction", "units" => "m/s")
-    w_attr = Dict("longname" => "Velocity in the z-direction", "units" => "m/s")
-    T_attr = Dict("longname" => "Temperature", "units" => "K")
-    S_attr = Dict("longname" => "Salinity", "units" => "g/kg")
+Construct a `NetCDFOutputWriter` that writes `label, field` pairs in `outputs` (which can be a
+`Dict` or `NamedTuple`) to a NC file, where `label` is a symbol that labels the output and
+`field` is a field from the model (e.g. `model.velocities.u`).
 
-    filepath = joinpath(fw.dir, filename(fw, "", iteration))
+Keyword arguments
+=================
 
-    if fw.async
-        println("[Worker $(Distributed.myid()): NetCDFOutputWriter] Writing fields to disk: $filepath")
-    end
+    - `filename::String`    : Directory to save output to. Default: "." (current working directory).
+    - `frequency::Int`      : Save output every `n` model iterations.
+    - `interval::Int`       : Save output every `t` units of model clock time.
+    - `clobber::Bool`       : Remove existing files if their filenames conflict. Default: `false`.
+    - `compression::Int`    : Determines the compression level of data (0-9, default 0)
+    - `globalattrib::Dict`  : Dict of model properties to save with every file (deafult: Dict())
+    - `outputattrib::Dict`  : Dict of attributes to be saved with each field variable (reasonable
+                              defaults are provided for velocities, temperature, and salinity)
+    - `slice_kw`            : `dimname = Union{OrdinalRange, Integer}` will slice the dimension `dimname`.
+                              All other keywords are ignored.
+                              e.g. `xC = 3:10` will only produce output along the dimension `xC` between
+                              indices 3 and 10 for all fields with `xC` as one of their dimensions.
+                              `xC = 1` is treated like `xC = 1:1`.
+                              Multiple dimensions can be sliced in one call. Not providing slices writes
+                              output over the entire domain.
+"""
+
+function NetCDFOutputWriter(model, outputs; interval=nothing, frequency=nothing, filename=".",
+                        clobber=true, globalattrib=Dict(), outputattrib=nothing, compression=0, slice_kw...)
```

Comment:
probably want underscores and full names here, eg `global_attributes` and `output_attributes`

Reply:
Fixed.